_infra/helm/locust/locustfiles/locustfile.py

```diff
@@ -441,32 +444,52 @@ def on_test_stop(environment, **kwargs):
             gcs.upload(file_name=history, file=h.read())
 
 
-class FrontstageTasks(SequentialTaskSet):
-    create_message_link = None
-    view_message_thread_link = None
+class Mixins:
+    csrf_token = None
+    auth_cookie = None
+
+    def get(self, url: str, expected_response_text=None):
+        with self.client.get(url=url, allow_redirects=False, catch_response=True) as response:
+
+            if response.status_code != 200:
+                error = f"Expected a 200 but got a {response.status_code} for url {url}"
+                response.failure(error)
+                self.interrupt()
+
+            if expected_response_text and expected_response_text not in response.text:
+                error = f"response text ({expected_response_text}) isn't in returned html"
+                response.failure(error)
+                self.interrupt()
+
+            return response
+
+    def post(self, url: str, data: dict = {}):
+        data['csrf_token'] = self.csrf_token
+        with self.client.post(url=url, data=data, allow_redirects=False, catch_response=True) as response:
+
+            if response.status_code != 302:
+                error = f"Expected a 302 but got a ({response.status_code}) for url {url} and data {data}"
+                response.failure(error)
+                self.interrupt()
+
+            return response
+
+
+class FrontstageTasks(TaskSet, Mixins):
 
     def on_start(self):
-        self.login()
-
-    def login(self):
-        sample_unit_ref = '499' + format(str(random.randint(0, respondents)), "0>8s")
-        data = {'username': sample_unit_ref + "@test.com", 'password': os.getenv('test_respondent_password')}
-        response = self.client.post("/sign-in/?next=", data=data, catch_response=True, allow_redirects=False)
-        if response.status_code != 302:
-            self.interrupt()
-        logger.info("status_code: %s", response.status_code)
+        self.sign_in()
+
+    def sign_in(self):
+        response = self.get(url="/sign-in", expected_response_text="Sign in")
+        self.csrf_token = _capture_csrf_token(response.content.decode('utf8'))
+
+        response = self.post("/sign-in", data=_generate_random_respondent())
         self.auth_cookie = response.cookies['authorization']
-        logger.info("auth_cookie: %s", self.auth_cookie)
 
-    @task(1)
-    def surveys_todo(self):
-        with self.client.get("/surveys/todo", cookies={"authorization": self.auth_cookie},
-                             catch_response=True) as response:
-            if 'Sign in' in response.text:
-                response.failure("Not logged in")
-            if 'To do' not in response.text:
-                response.failure(
-                    "To do tab not displayed after login")
+    @task
+    def todo(self):
```

Comment:
Long term this isn't how we want to do this, as it should be stored in a request file with data, but for now this is fine